File: app/models/article.rb
# frozen_string_literal: true

class Article < ApplicationRecord
  include Visible

  has_many :comments, dependent: :destroy

  validates :title, presence: true
  validates :body, presence: true, length: { minimum: 10 }

  def self.public_article_count
    Article.where(status: 'public').count
  end
end

File: frontend/src/components/Article.js
const Articles = ({ articles }) => {
  return (
    <>
      {articles.map((article) => {
        return (
          <div key={article.id}>
            <p>{article.title}</p>
            <p>{article.body}</p>
          </div>
        );
      })}
    </>
  );
};

export default Articles;
